### PR TITLE
Fix issue 26965 transaction scope crash

### PR DESCRIPTION
**Pull request by keeratsingh**
Fixes GitHub issue: #26965

**Problem Description:**
The culprit here is the platform specific if block, it only initializes the __**transactedConnectionPool**_ object for Windows and not for Linux/Mac, hence it later throws a NullReferenceException while closing the connection.

**Affected Tests:**
System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest

**Review:**
Requesting review @saurabh500 @geleems @corivera 

**Before Changes**
```bash
keerats@ubuntu:/mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests$ ~/dotnetcore/corefx/Tools/msbuild.sh /t:RebuildAndTest /p:ConfigurationGroup=Release /p:XunitOptions="-class System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest"
Microsoft (R) Build Engine version 15.5.179.9764 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Address -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/Address/netstandard/Address.dll
  Circle -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/Circle/netstandard/Circle.dll
  Shapes -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/Shapes/netstandard/Shapes.dll
  Utf8String -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/Utf8String/netstandard/Utf8String.dll
  System.Data.SqlClient.ManualTesting.Tests -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/System.Data.SqlClient.ManualTesting.Tests/netstandard/System.Data.SqlClient.ManualTesting.Tests.dll
  ----- start 13:33:50 =============== To repro directly: =====================================================
  pushd /mnt/d/corefx/bin/tests/System.Data.SqlClient.ManualTesting.Tests/netcoreapp-Linux-Release-x64
  chmod +x /mnt/d/corefx/bin/testhost/netcoreapp-Linux-Release-x64//dotnet
  /mnt/d/corefx/bin/testhost/netcoreapp-Linux-Release-x64//dotnet xunit.console.netcore.exe System.Data.SqlClient.ManualTesting.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -class System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest  -notrait category=OuterLoop -notrait category=failing
  popd
  ===========================================================================================================
  /mnt/d/corefx/bin/tests/System.Data.SqlClient.ManualTesting.Tests/netcoreapp-Linux-Release-x64 /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests
  xUnit.net console test runner (64-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.
  
  Discovering: System.Data.SqlClient.ManualTesting.Tests
  Discovered:  System.Data.SqlClient.ManualTesting.Tests
  Starting:    System.Data.SqlClient.ManualTesting.Tests
     System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest.TestAutoEnlistment_TxScopeComplete [FAIL]
        System.NullReferenceException : Object reference not set to an instance of an object.
        Stack Trace:
           /mnt/d/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs(834,0): at System.Data.ProviderBase.DbConnectionPool.DeactivateObject(DbConnectionInternal obj)
           /mnt/d/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs(1458,0): at System.Data.ProviderBase.DbConnectionPool.PutObject(DbConnectionInternal obj, Object owningObject)
           /mnt/d/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionInternal.cs(451,0): at System.Data.ProviderBase.DbConnectionInternal.CloseConnection(DbConnection owningObject, DbConnectionFactory connectionFactory)
           /mnt/d/corefx/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs(505,0): at System.Data.SqlClient.SqlConnection.Close()
           /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/TransactionEnlistmentTest.cs(59,0): at System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest.TestCase_AutoEnlistment_TxScopeComplete()
           /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/TransactionEnlistmentTest.cs(197,0): at System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest.RunTestFormat(Action testCase)
           /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/TransactionEnlistmentTest.cs(185,0): at System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest.RunTestSet(Action TestCase)
     System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest.TestAutoEnlistment_TxScopeNonComplete [FAIL]
        System.NullReferenceException : Object reference not set to an instance of an object.
        Stack Trace:
           /mnt/d/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs(834,0): at System.Data.ProviderBase.DbConnectionPool.DeactivateObject(DbConnectionInternal obj)
           /mnt/d/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs(1458,0): at System.Data.ProviderBase.DbConnectionPool.PutObject(DbConnectionInternal obj, Object owningObject)
           /mnt/d/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionInternal.cs(451,0): at System.Data.ProviderBase.DbConnectionInternal.CloseConnection(DbConnection owningObject, DbConnectionFactory connectionFactory)
           /mnt/d/corefx/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs(505,0): at System.Data.SqlClient.SqlConnection.Close()
           /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/TransactionEnlistmentTest.cs(84,0): at System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest.TestCase_AutoEnlistment_TxScopeNonComplete()
           /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/TransactionEnlistmentTest.cs(197,0): at System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest.RunTestFormat(Action testCase)
           /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/TransactionEnlistmentTest.cs(185,0): at System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest.RunTestSet(Action TestCase)
  Finished:    System.Data.SqlClient.ManualTesting.Tests
  
  === TEST EXECUTION SUMMARY ===
/mnt/d/corefx/Tools/tests.targets(492,5): warning :    System.Data.SqlClient.ManualTesting.Tests  Total: 5, Errors: 0, Failed: 2, Skipped: 0, Time: 1.507s [/mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/System.Data.SqlClient.ManualTesting.Tests.csproj]
  /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests
  ----- end 13:33:52 ----- exit code 1 ----------------------------------------------------------
/mnt/d/corefx/Tools/tests.targets(492,5): warning MSB3073: The command "/mnt/d/corefx/bin/tests/System.Data.SqlClient.ManualTesting.Tests/netcoreapp-Linux-Release-x64/RunTests.sh /mnt/d/corefx/bin/testhost/netcoreapp-Linux-Release-x64/" exited with code 1. [/mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/System.Data.SqlClient.ManualTesting.Tests.csproj]
/mnt/d/corefx/Tools/tests.targets(500,5): error : One or more tests failed while running tests from 'System.Data.SqlClient.ManualTesting.Tests' please check /mnt/d/corefx/bin/tests/System.Data.SqlClient.ManualTesting.Tests/netcoreapp-Linux-Release-x64/testResults.xml for details! [/mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests/System.Data.SqlClient.ManualTesting.Tests.csproj]
keerats@ubuntu:/mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests$ 

```

**After Changes**
```bash
keerats@ubuntu:/mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests$ ~/dotnetcore/corefx/Tools/msbuild.sh /t:RebuildAndTest /p:ConfigurationGroup=Release /p:XunitOptions="-class System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest"
Microsoft (R) Build Engine version 15.5.179.9764 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Circle -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/Circle/netstandard/Circle.dll
  Address -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/Address/netstandard/Address.dll
  Shapes -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/Shapes/netstandard/Shapes.dll
  Utf8String -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/Utf8String/netstandard/Utf8String.dll
  System.Data.SqlClient.ManualTesting.Tests -> /mnt/d/corefx/bin/AnyOS.AnyCPU.Release/System.Data.SqlClient.ManualTesting.Tests/netstandard/System.Data.SqlClient.ManualTesting.Tests.dll
  ----- start 13:40:09 =============== To repro directly: =====================================================
  pushd /mnt/d/corefx/bin/tests/System.Data.SqlClient.ManualTesting.Tests/netcoreapp-Linux-Release-x64
  chmod +x /mnt/d/corefx/bin/testhost/netcoreapp-Linux-Release-x64//dotnet
  /mnt/d/corefx/bin/testhost/netcoreapp-Linux-Release-x64//dotnet xunit.console.netcore.exe System.Data.SqlClient.ManualTesting.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -class System.Data.SqlClient.ManualTesting.Tests.TransactionEnlistmentTest  -notrait category=OuterLoop -notrait category=failing
  popd
  ===========================================================================================================
  /mnt/d/corefx/bin/tests/System.Data.SqlClient.ManualTesting.Tests/netcoreapp-Linux-Release-x64 /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests
  xUnit.net console test runner (64-bit .NET Core)
  Copyright (C) 2014 Outercurve Foundation.
  
  Discovering: System.Data.SqlClient.ManualTesting.Tests
  Discovered:  System.Data.SqlClient.ManualTesting.Tests
  Starting:    System.Data.SqlClient.ManualTesting.Tests
  Finished:    System.Data.SqlClient.ManualTesting.Tests
  
  === TEST EXECUTION SUMMARY ===
     System.Data.SqlClient.ManualTesting.Tests  Total: 5, Errors: 0, Failed: 0, Skipped: 0, Time: 1.680s
  /mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests
  ----- end 13:40:11 ----- exit code 0 ----------------------------------------------------------
keerats@ubuntu:/mnt/d/corefx/src/System.Data.SqlClient/tests/ManualTests$ 
```

### CODE REVIEW

File: src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs [hunk 1]
```diff
@@ -413,10 +413,7 @@ internal WaitHandle[] GetHandles(bool withCreate)
 
             _objectList = new List<DbConnectionInternal>(MaxPoolSize);
 
-            if (ADP.IsPlatformNT5)
-            {
-                _transactedConnectionPool = new TransactedConnectionPool(this);
-            }
+            _transactedConnectionPool = new TransactedConnectionPool(this); // initialize irrespective of platform
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs]: In AdapterUtil.cs `ADP.IsPlatformNT5` is not used anymore by any other part of the code. Can you remove these unused attributes as well?

Comment (keeratsingh) [reply src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPool.cs]: Done

File: src/Common/src/System/Data/Common/AdapterUtil.cs [hunk 2]
```diff
@@ -486,11 +486,9 @@ internal static Exception InvalidSeekOrigin(string parameterName)
         }
 
         internal static readonly bool IsWindowsNT = (PlatformID.Win32NT == Environment.OSVersion.Platform);
```
Comment (saurabh500) [inline src/Common/src/System/Data/Common/AdapterUtil.cs]: Can this line `IsWindowsNT` be removed as well?

Comment (keeratsingh) [reply src/Common/src/System/Data/Common/AdapterUtil.cs]: Yes, certainly :)

**Comment by saurabh500** [comment]
Thanks. Ready to merge when CI is green

**Comment by saurabh500** [comment]
@dotnet-bot Test Windows x86 Release Build Please

**Comment by ahsonkhan** [comment]
@dotnet-bot test Linux x64 Release Build

https://mc.dot.net/#/user/keeratsingh/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/a6cb6988c6f98666e7100eb052dde530172d476a/workItem/System.Net.Security.Tests/wilogs
Ubuntu.1404.Amd64.Open:Release-x64
```text
2018-02-28 00:15:52,380: INFO: proc(54): run_and_log_output: Output: Starting:    System.Net.Security.Tests
2018-02-28 00:15:55,468: INFO: proc(54): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/8374ee34-8f38-4620-9296-7a47873dc745/Work/0dbd7862-1367-43a5-a5bc-ab2d11207474/Unzip/RunTests.sh: line 88: 20802 Aborted                 (core dumped) $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Net.Security.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=OuterLoop -notrait category=failing
2018-02-28 00:15:55,508: INFO: proc(54): run_and_log_output: Output: Dumpling has not been installed yet. Please run "DumplingHelper.py install_dumpling" before collect_dumps.
2018-02-28 00:15:55,511: INFO: proc(54): run_and_log_output: Output: ~/dotnetbuild/work/8374ee34-8f38-4620-9296-7a47873dc745/Work/0dbd7862-1367-43a5-a5bc-ab2d11207474/Unzip
2018-02-28 00:15:55,513: INFO: proc(54): run_and_log_output: Output: ----- end 00:15:55 ----- exit code 134 ----------------------------------------------------------
2018-02-28 00:15:55,515: INFO: proc(54): run_and_log_output: Output: Unable to find executable corerun
2018-02-28 00:15:55,517: INFO: proc(54): run_and_log_output: Output: Copying core file core to /tmp/coredumps
2018-02-28 00:15:55,721: INFO: proc(58): run_and_log_output: Exit Code: 134
```
cc @davidsh, @wfurt, @bartonjs

**Pull request merged**